sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -429,5 +534,88 @@ protected EventProcessor()
 
             throw new NotImplementedException();
         }
+
+        /// <summary>
+        ///   An virtual <see cref="StorageManager" /> instance that delegates calls to the
+        ///   <see cref="EventProcessor{TPartition}" /> to which it is associated.
+        /// </summary>
+        ///
+        private class DelegatingStorageManager : StorageManager
```

Comment:
The only reason we are wrapping the service calls in a `StorageManager` is because we need to pass it to the `PartitionLoadBalancer`, correct?

Reply:
More or less.  The indirection keeps the load balancer from having a circular relationship with the processor where they both have references to each other.   I also see the storage manager as a nice way to wrap and test storage stuff as a separate unit of responsibility.   I can see that being helpful should we roll out other stores in the future.